module/zfs/dmu_send.c

```diff
@@ -642,6 +642,7 @@ dump_freeobjects(dmu_send_cookie_t *dscp, uint64_t firstobj, uint64_t numobjs)
 	 * send FREEOBJECTS records for object IDs which cannot exist on the
 	 * receiving side.
 	 */
+	ASSERT3U(numobjs, !=, 0);
 	if (maxobj > 0) {
 		if (maxobj < firstobj)
```

Comment:
I think that `numobjs` can become zero, if `firstobj == maxobj`.  That doesn't seem useful, seems like in that case we should return.  i.e.:
```suggestion
		if (firstobj >= maxobj)
```